source/paddle.js

```diff
@@ -889,6 +922,618 @@ paddle.AttributeType = {
+            description.set(attrName, paddle.IR.Utility.formatOutputAttr(attrName, attrValue));
+        }
+
+        return description;
+    }
+};
+
+paddle.IR.OpInfoP = class extends paddle.IR.OpInfo {
+
+    init(op) {
+        const [name] = op.A.slice(3);
+        this._name = name;
+        this._type = paddle.IR.Utility.getCompressOp(this._type);
+        this._fullName = this._type;
+        op.OA = [...op.OA, ...op.A];
+    }
+
+    get type() {
+        return this._type;
+    }
+
+    get name() {
+        return this._type;
+    }
+
+    get namePrefix() {
+        return this._name;
+    }
+
+    get fullName() {
+        return this._fullName;
+    }
+
+    getAttr(idx, value) {
+        let attrName = '';
+        let attrType = '';
+        let attrValue = '';
+        switch (idx) {
+            case '0':
+                attrName = 'is_distributed';
+                attrType = paddle.IR.Utility.getType('a_bool');
+                break;
+            case '1':
+                attrName = 'is_parameter';
+                attrType = paddle.IR.Utility.getType('a_bool');
+                break;
+            case '2':
+                attrName = 'need_clip';
+                attrType = paddle.IR.Utility.getType('a_bool');
+                break;
+            case '3':
+                attrName = 'name';
+                attrType = paddle.IR.Utility.getType('a_str');
+                break;
+            default:
+                break;
+        }
+        attrValue = attrType === paddle.IR.Utility.getType('a_bool') ? value === 1 : value;
+        return [attrName, attrType, attrValue];
+    }
+
+    getOutputAttr(idx, outputAttr) {
+        const persistable = outputAttr[0] === 1;
+        const stop_gradient = outputAttr[1] === 1;
+        const trainable = outputAttr[2] === 1;
+        const is_distributed = outputAttr[3] === 1;
+        const is_parameter = outputAttr[4] === 1;
+        const need_clip = outputAttr[5] === 1;
+        return new Map([
+            ['persistable', paddle.IR.Utility.formatOutputAttr('persistable', persistable)],
+            ['stop_gradient', paddle.IR.Utility.formatOutputAttr('stop_gradient', stop_gradient)],
+            ['trainable', paddle.IR.Utility.formatOutputAttr('trainable', trainable)],
+            ['is_distributed', paddle.IR.Utility.formatOutputAttr('is_distributed', is_distributed)],
+            ['is_parameter', paddle.IR.Utility.formatOutputAttr('is_parameter', is_parameter)],
+            ['need_clip', paddle.IR.Utility.formatOutputAttr('need_clip', need_clip)],
+            ['name', paddle.IR.Utility.formatOutputAttr('name', outputAttr[6])]
+        ]);
+    }
+};
+
+paddle.IR.Attr = class {
+
+    constructor(idx, value, opInfo) {
+        const [attrName, attrType, attrValue] = opInfo.getAttr(idx, value);
+        this.name = attrName;
+        this.type = attrType;
+        this.value = attrValue;
+    }
+};
+
+paddle.IR.Input = class {
```

Comment:
`paddle.IR.Input` and `paddle.IR.Output` have been removed, instead, `createInput` and `createOutput` have been added within `paddle.IR.Op` ~ `values.map` may not be so convenient, because the `values` are parsed within the `paddle.IR.Op` and the sub regions ... ... Do you have any better suggestions? Thanks!